### ISSUE TITLE
False positive "Join condition is not allowed for cross joins."

### ISSUE THREAD

**Issue by hfhchan-plb** [sbdchd/squawk#718]
The following query triggers a "Join condition is not allowed for cross joins." error:

```sql
INSERT INTO ...
SELECT ...
FROM ...
CROSS JOIN ...
ON CONFLICT DO NOTHING;
```

The ON clause is getting read as part of the CROSS JOIN, but is actually part of the INSERT query
